### PR TITLE
Remove derp section from ClientConnectivity model

### PR DESCRIPTION
**Pull request by heretic098**
# Proposed Changes

Remove the requirement for the tailscale API to return the derp information in the ClientConnectivity section of the response as it has been removed in https://github.com/tailscale/tailscale/commit/b7988b382584d7129b15da7590280f5400123891.

Create a default empty list to hold tags in case none are returned from the API. This avoids issues returning information where some devices have no tags defined. In this case with the current implementation you see errors like this:

```
devices.XXXXXXXXXXX.tags
  Field required [type=missing, input_value={'addresses': ['XXX.YY.ZZ...: True, 'upnp': False}}}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.0.3/v/missing
```

## Related Issues

#427 

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

**Comment by heretic098** [comment]
@frenck , I don't seem to be able to resolve the issue with the labels myself. Any assistance appreciated.

File: src/tailscale/models.py [hunk 1]
```diff
@@ -44,9 +42,9 @@ class Device(BaseModel):
     client_version: str = Field(..., alias="clientVersion")
     update_available: bool = Field(..., alias="updateAvailable")
     os: str
-    created: datetime | None
-    last_seen: datetime | None = Field(..., alias="lastSeen")
-    tags: list[str] | None
+    created: Optional[datetime]
+    last_seen: Optional[datetime] = Field(..., alias="lastSeen")
```
Comment (frenck) [inline src/tailscale/models.py:46]:
```suggestion
    created: datetime | None
    last_seen: datetime | None = Field(..., alias="lastSeen")
```

File: src/tailscale/models.py [hunk 2]
```diff
@@ -23,11 +23,9 @@ class ClientSupports(BaseModel):
 class ClientConnectivity(BaseModel):
     """Object holding Tailscale device information."""
 
-    endpoints: list[str] = Field(default_factory=list)
-    derp: str
-    mapping_varies_by_dest_ip: bool | None = Field(
-        None,
-        alias="mappingVariesByDestIP",
+    endpoints: List[str] = Field(default_factory=list)
```
Comment (frenck) [inline src/tailscale/models.py:26]:
```suggestion
    endpoints: list[str] = Field(default_factory=list)
```

File: src/tailscale/models.py [hunk 3]
```diff
@@ -23,11 +23,9 @@ class ClientSupports(BaseModel):
...
+    mapping_varies_by_dest_ip: Optional[bool] = Field(
```
Comment (frenck) [inline src/tailscale/models.py:27]:
```suggestion
    mapping_varies_by_dest_ip: bool | None = Field(
```

File: src/tailscale/models.py [hunk 4]
```diff
@@ -46,7 +44,7 @@ class Device(BaseModel):
     os: str
     created: datetime | None
     last_seen: datetime | None = Field(..., alias="lastSeen")
-    tags: list[str] | None
+    tags: List[str] | None = Field(default=[])
```
Comment (frenck) [inline src/tailscale/models.py:47]:
```suggestion
    tags: list[str] | None = Field(default=[])
```

File: src/tailscale/models.py [hunk 5]
```diff
@@ -24,10 +24,8 @@ class ClientConnectivity(BaseModel):
     """Object holding Tailscale device information."""
 
     endpoints: list[str] = Field(default_factory=list)
-    derp: str
     mapping_varies_by_dest_ip: bool | None = Field(
-        None,
-        alias="mappingVariesByDestIP",
+        None, alias="mappingVariesByDestIP"
```
Comment (frenck) [inline src/tailscale/models.py:28]:
```suggestion
        None,
        alias="mappingVariesByDestIP",
```

**Comment by frenck** [comment]
Sorry for the mess in the changes above. I messed up the merge :(

**Review by frenck — APPROVED** [review]
Thanks, @heretic098 👍 

../Frenck

**Pull request merged**